### ISSUE TITLE
Cannot connect to OrbStack .local host anymore

### ISSUE THREAD

**Issue by eproxus** [TablePlus/TablePlus#3407]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

   Server:
   
   > PostgreSQL 17.0 (Debian 17.0-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
   
   Driver: _No idea what driver version TablePlus is using..._

3. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
   
   > Version 6.1.6 (570)

5. The steps to reproduce this issue:
	1. Start a Postgres container in OrbStack (e.g. the official `postgres:17`):
	   ```
       docker run --interactive --tty --name postgres_container --env POSTGRES_PASSWORD=foobar --publish 55432:5432 postgres:17
       ```
       Note the hostname in the OrbStack GUI (e.g. `postgres_container.orb.local`)
	3. Verify it is reachable
	   ```
	   $ traceroute postgres_container.orb.local
       traceroute to postgres_container.orb.local (192.168.215.2), 64 hops max, 40 byte packets
       1  198.19.248.65 (198.19.248.65)  2.542 ms  0.595 ms  0.273 ms
       2  192.168.215.2 (192.168.215.2)  0.441 ms  0.836 ms  0.453 ms
       ```
    4. Verify that e.g. `pgcli` works:
       ```
       » pgcli --host postgres_container.orb.local --user postgres --dbname postgres
       Password for postgres:
       Server: PostgreSQL 17.0 (Debian 17.0-1.pgdg120+1)
       Version: 4.1.0
       Home: http://pgcli.com
       postgres>
       ```
	5. Try to connect to `postgres_container.orb.local` with TablePlus

If I try this connection with TablePlus I get the following error:

<img width="612" alt="Screenshot 2024-10-07 at 15 14 08" src="https://github.com/user-attachments/assets/54eaa4a2-7456-43dc-8466-35adb1066ca1">

Using a normal localhost IP works:

<img width="612" alt="Screenshot 2024-10-07 at 15 13 50" src="https://github.com/user-attachments/assets/71031a10-456c-4c39-8b4c-3e8e6e2462d8">

This used to work on previous versions of TablePlus, and since other tools work I at least assume for now the problem is with TablePlus.

Let me know if you need anything else.

**Comment by huyphams** [comment]
You mean other tool  which runs on CLI works @eproxus ? I believe it works because the CLI can load the env which TablePlus could not load. You may need to add the postgres_container.orb.local to host.

it works on CLI does not mean it can be loaded in TablePlus.

**Comment by huyphams** [comment]
> Driver: No idea what driver version TablePlus is using...

I believe that we did not change any core functions which could lead to this issue.

**Comment by huyphams** [comment]
Ah, nice! I think it’s a new requirement. I’ve started seeing other applications asking for local network discovery permission.

Actually, I'm not a fan of this. I don't want an app that asks for things it doesn't need or could confuse users, but it's good if it can point the user to the documentation.

Again thank you!

**Comment by eproxus** [comment]
Yeah, I mean the OS will ask the first time when an app tries to connect. It was just in my case that I probably missed it somehow (which I guess happens easily). The downside is of course that then the OS can’t notify me again since it would be spammy.

**Comment by iloveitaly** [comment]
I'm not seeing the option to enable TablePlus in that settings area and it's failing to connect to an `*orb.local` domain :( Any ideas?

**Comment by iloveitaly** [comment]
This is occurring intermittently for me. I restarted my computer and now it's occurring, but before I was fine.

**Comment by iloveitaly** [comment]
Ok, strange development here:

1. If I attempt to use a local domain such as `postgres.cosma.orb.local` it fails
2. If I run `dns-sd -Q postgres.cosma.orb.local` and use the resulting IP, it works
